ch5-catdir/controller.h

```diff
@@ -0,0 +1,37 @@
+/*************************************************
+ * controller.h - Definision of class Controller *
+ *************************************************/
+
+#pragma once
+
+#include <cstdlib>
+#include <iosfwd>
+#include <string_view>
+
+/*
+ * Controller - the class that control the execution of the program
+ */
+
+class Controller
+{
+public:
+    Controller(int argc, char* argv[], std::ostream& out, std::ostream& err);
+    ~Controller();
+    Controller(const Controller&)            = delete;
+    Controller& operator=(const Controller&) = delete;
+    int run();
+
+    static std::ostream& error();
+
+private:
+    void process(const std::string_view& path);
+
+    static Controller* m_instance; // static pointer to the created controller
```

Comment:
Maybe you want to use `inline`? Also, I prefer to explicitly init to `nullptr` instead of counting on the fact it's `static`

Reply:
I did both.